Define the role and organization of the foundation (#32)
NixOS/foundation: role_and_responsibilities.md

# Roles and responsibilities of the Foundation and the Foundation board

## Role of the Foundation

> The foundation is the legal entity that backs the project from an organizational, material, and social point of view.

Reviewer: ```suggestion
The foundation is the legal entity that backs the project from an organizational, fiduciary, material, and social point of view.
```

Its goal is to set the Nix community up for long term and scalable success.

This means:

- Ensure the core infrastructure of the ecosystem
  - Provide the necessary technical infrastructure (GitHub, Hydra, Binary cache, etc..)
  - Provide the necessary financial and legal support
    - Register and protect the trademark
    - Handle the funding of the community, by both accepting external funding and by redistributing it where needed
  - Provide the necessary human infrastructure
    - Host and empower the community teams

>     - Provide a conflict moderation and resolution path

Reviewer 2: This doesn't exactly match the rest of the mission statement. The foundation cannot provide it, but support and arrange for people to get together. Conflict moderation is the moderation team's job, and you already covered the escalation capacity (which would be part of the board's job) at the bottom.

  - Ensure the necessary security infrastructure

- Support the growth and functioning of the community
  - Represent Nix in the world
    - Create and maintain partnerships
    - Organize or support Nix events (NixCon, Summer of Nix, hackathons, …)
  - Maintain the community in a healthy state

>     - Enforce some needed rules to keep the community an open and welcoming place (with or without a code of conduct)

Reviewer 2: ```suggestion
    - Enforce some needed rules to keep the community an open and welcoming place
```

No need to be that specific here.

>     - Run regular community surveys to take the pulse of the community and share back the results

Reviewer 2: ```suggestion
```

This is the marketing team's job.

Reviewer 2: Maybe put it a bit more generally: gather data to take the pulse of the community and communicate the results.

Reviewer 3: We'll keep having conversations and updating the document. This doesn't need to be set in stone.

These responsibilities are carried out by the different teams that form the community (infrastructure team, security team, marketing team, community team, etc.).

## Role of the board

_Empower the community and stay out of the way_

> The role of the board is to support the day-to-day operations of the community, help to structure it, and serve as a last-resort backup when things go wrong.

Reviewer 3: ```suggestion
The role of the board is to provide operational and organizational support to the NixOS project and community.

The board is not responsible for technical leadership, as this is left to the individual community teams.
```
I would like it if it was a bit more explicit that the board doesn't provide technical leadership, so that members can turn to the right people when looking to participate in the ecosystem.

Reviewer: I would even expand that to make it clear that -> The board is not responsible for technical leadership, decisions or direction

Author: Indeed. I've made that more explicit in ad5a6a779084159eb27d83fb6482357e434af667

In particular, its responsibilities are to:

- Handle the administrative, legal and financial tasks of the Foundation
- Serve as an interface between the community and the corporate/governmental world

>   - Handle sponsoring and donations

Reviewer: - Build and maintain beneficial and collaborative relationships (an example here would be Github and maybe other foundations for collaboration)

Author: Added in ba75a95f693a285c860fd98bc5c119bf632056c8

> - Organize the team structure

Reviewer 2: ```suggestion
- Provide a framework for teams to self-organise
```

This aligns more with the supportive role you're outlining. You're not organising, you're helping people get productive.

- Fund community events and efforts
- Unblock things that would be stuck otherwise
  - Arbiter in case of conflicts

>   - Serve as a backup for critical tasks when needed

Reviewer 2: What's missing here is handing out permissions and credentials. Someone has to hold and manage the keys.

Author: Added in 4b6d915 (although I'm not entirely sure I've put it in the best possible place)

Reviewer 3: Ideally, the Foundation members do not hold any critical security tokens, but have a good working relationship with and can delegate those tasks to the Infra team. Otherwise, it will be difficult for the infra team to guarantee the security of the infrastructure as a whole.

Author: @zimbatm I would argue that the infra team needs in practice to have access to _most_ systems, but it shouldn't have the responsibility of delegating the access to other teams. Maybe that shouldn't be the foundation board either though, but then we don't have anybody in a good position for it.

Reviewer 3: The board would still hold the power to instruct the infra team to delegate access, either directly, or by delegating some of that access further down to the community. For example by setting up an election system.

But this is more of a future statement. The current infra team needs more people and a better interface.
